video_xblock/static/js/studio-edit.js

```diff
@@ -457,53 +457,70 @@ function StudioEditableXBlock(runtime, element) {
+     *  currentLiTag (Object): DOM element containing information on an uploaded subtitle.
      */
-    function successHandler(response, statusText, xhr, fieldName, lang, label, currentLiTag) {
+    function successHandler(event, response, statusText, xhr, fieldName, lang, label, currentLiTag) {
         var url = '/' + response['asset']['id'];
-        var regExp = /.*@(.+\..+)/;
-        var filename = regExp.exec(url)[1];
+        // User can upload a file without extension
+        var filename = $fileUploader[0].files[0].name;
         var downloadUrl = downloadTranscriptHandlerUrl + '?' + url;
-        if (fieldName == 'handout') {
-            var $parentDiv = $('.file-uploader', element);
+        var successMessage = 'File "' + filename + '" uploaded successfully';
+        var $parentDiv;
+        var downloadUrlServer;
+        var defaultTranscript;
+        var isValidated = validateTranscriptFile(event, fieldName, filename, $fileUploader);
+        if (fieldName == 'handout' && isValidated) {
+            $parentDiv = $('.file-uploader');
             $('.download-setting', $parentDiv).attr('href', downloadUrl).removeClass('is-hidden');
             $('a[data-change-field-name=' + fieldName + ']').text('Replace');
-            showUploadStatus($parentDiv, filename);
+            displayStatusCaptions('success', successMessage, $parentDiv);
             $('input[data-field-name=' + fieldName + ']').val(url).change();
-        } else {
+        } else if (fieldName == 'transcripts' && isValidated) {
             pushTranscript(lang, label, url, '', transcriptsValue);
             $('.add-transcript').removeClass('is-disabled');
             $('input[data-field-name=' + fieldName + ']').val(JSON.stringify(transcriptsValue)).change();
             $(currentLiTag).find('.upload-transcript').text('Replace');
             $(currentLiTag).find('.download-transcript')
                 .removeClass('is-hidden')
                 .attr('href', downloadUrl);
-            showUploadStatus($(currentLiTag), filename);
-            // Affect default transcripts
-            // Update respective enabled transcript with an external url from a newly created standard transcript
-            var downloadUrlServer = $('.list-settings-buttons .upload-setting.upload-transcript[data-lang-code=' + lang + ']')
+            displayStatusTranscripts('success', successMessage, currentLiTag);
+            // Affect default transcripts: update a respective enabled transcript with an external url
+            // of a newly created standard transcript
+            downloadUrlServer =
+                $('.list-settings-buttons .upload-setting.upload-transcript[data-lang-code=' + lang + ']')
                 .siblings('a.download-transcript.download-setting').attr('href');
-            var defaultTranscript= {'lang': lang, 'label': label, 'url': downloadUrlServer};
+            defaultTranscript= {'lang': lang, 'label': label, 'url': downloadUrlServer};
```

Comment:
Wow, I thought eslint should've caught such things. [Space-infix-rule](http://eslint.org/docs/rules/space-infix-ops) seems to be the rule for such things but, apparently, it does not do the job.